File: pset1/cash/cash.c
#import <cs50.h>
#import <stdio.h>
#import <math.h>

int main(void)
{
    int coins[] = {25, 10, 5, 1};
    int coins_count = 4;
    int result = 0;

    // Get user input
    float number;
    do
    {
        number = get_float("Change owed: ");
    }
    while (number < 0);

    // Convert to cents
    int cents = round(number * 100);

    // Calculate how many coins to return
    float coins_quantity;
    for (int i = 0; i < coins_count; i ++)
    {
        coins_quantity = cents / coins[i];
        if (coins_quantity > 0)
        {
            cents -= coins[i] * coins_quantity;
            result += coins_quantity;
        }
    }
    printf("%d\n", result);
}

File: pset6/cash/cash.py
from cs50 import get_float

# Prompt user for change value
while True:
    change = get_float("Number: ")
    if change >= 0:
        break

# Round input value and convert to cents
change = round(change, 2) * 100
coins = [25, 10, 5, 1]
total_coins = 0

# Loop over each available coin
for coin in coins:
    coins_number, change = divmod(change, coin)
    total_coins += int(coins_number)

print(total_coins)